File: 2017/Animators/ObjectAnimator/app/src/main/java/me/mehdi/objectanimator/MainActivity.java
package me.mehdi.objectanimator;

import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button mButton;
    ObjectAnimator mScaleX;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mButton = findViewById(R.id.start);
        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View button) {
                mScaleX = ObjectAnimator.ofFloat(button, "scaleX", 5);
                mScaleX.setDuration(1000);
                mScaleX.setRepeatCount(ValueAnimator.INFINITE);
                mScaleX.setRepeatMode(ValueAnimator.REVERSE);
                mScaleX.start();
            }
        });
    }
}

File: Animators/app/src/main/java/com/example/animators/MainActivity.kt
package com.example.animators

import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.example.animators.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        binding.hello.setOnClickListener {button->
            val xScale = ObjectAnimator.ofFloat(button, View.SCALE_X, 5f)

            xScale.repeatCount = 3
            xScale.repeatMode = ValueAnimator.REVERSE
            xScale.duration = 1000

            xScale.start()
        }
    }
}